src/cargo/core/profiles.rs

```diff
@@ -539,10 +539,16 @@ pub enum ProfileRoot {
 
 /// Profile settings used to determine which compiler flags to use for a
 /// target.
-#[derive(Clone, Eq, PartialOrd, Ord, serde::Serialize)]
+#[derive(Clone, Eq, PartialOrd, Ord, serde::Serialize, derivative::Derivative)]
```

Comment:
When adding a dependency, please expand on why it is helpful and why it is worth taking on.

In particular, this is adding a dependency in place of hand-implementing one `Hash` and one `PartialEq`.  That seems like a fairly small scope that I would expect the dependency to be pulling a lot of weight.

Reply:
i don't feel code like that should be allowed to exist

Reply:
> i don't feel code like that should be allowed to exist

Please expand.

Communication is critical in open source, especially when maintains have limited availability.  So far the communication has been pretty minimal between this comment, this PR, and #11120.

I can make my own guesses and some of them might be quite valid but that is a one sided conversation with myself.  I'm wanting to understand your intentions and reasoning.